nerfstudio/models/nerfacto.py

```diff
@@ -321,15 +332,15 @@ def get_outputs(self, ray_bundle: RayBundle):
 
     def get_metrics_dict(self, outputs, batch):
         metrics_dict = {}
-        image = batch["image"].to(self.device)
+        image = batch["image"][..., :3].to(self.device)
```

Comment:
If blending is not used here as well, outputs["rgb"] as generated using the random noise background. This would decrease the metrics. Solution? Do not blend neither the RGB nor the target image, but do it inside of loss/metric computation?

Reply:
This wouldn't fix the metric problem as the rendered image still won't contain the original background as this wasn't used during training. For binary alpha masks, it could make sense only to compute the metrics using the pixels with alpha=1. This wouldn't make sense for non-binary alpha masks; however, I don't see a use case for having these.
What do you think about this?

Reply:
Why wouldn't it fix the problem? Both the prediction and the target would be RGBA images and for metrics computation, a default background colour would be used (e.g. black). Note, that the alpha channel for the prediction would be the accumulation. For loss computation, the matching random colour would be used. However, having the random colour for metrics computation would decrease LPIPS. This will also enable access to the original RGBA images should the user want to store them. What do you think of this proposal?

Reply:
Just to make sure that I understand, are you suggesting the following?
1. Use accumulation to mask the predicted images
2. Use alpha to mask the input images
3. Insert black background outside the masks
4. Compute the metrics based on these masked images

Reply:
I suggest the following:
1. predict RGBA images where alpha is the accumulation
2. for loss computation, perform blending with random background (if background_color=random) and compute the loss afterwards
3. for metric computation, if background_color==random perform blending with a non-random (black or configurable) colour. Then the metrics can be computed on the resulting RGB images.

I do not claim this to be the best approach. It is just a suggestion. What are your thoughts?

Reply:
Thanks for clarifying! I understand the suggestion a lot better now. 

I think this would be a reasonable approach. I'd say another viable option would be to use the original alpha masks to mask the images before computing the metrics. The first option includes the predicted alpha channel in the metric, while the second one does not. I'm not sure which one is more desirable as I have never used reconstruction metrics before. The second one would not require a configurable parameter. Do you have any preferences/ insights here?

Reply:
I would advise against using masks in metrics. 1) the alpha channel may not be binary 2) would be difficult to do for lpips and others

Reply:
Okay, these are good points. I will implement your suggestion then.

Reply:
Implemented this now. I added another method to the RGB renderer to avoid code duplication, but I'm not a massive fan of it. Let me know if you have suggestions for making this cleaner.

It does significantly improve the metrics and makes them meaningful.

Without the change:
![psnr_before](https://github.com/nerfstudio-project/nerfstudio/assets/53228351/62f25d28-d117-4674-bba7-6fdebf3a1276)

After the change:
![psnr_after](https://github.com/nerfstudio-project/nerfstudio/assets/53228351/36d7c90a-cfcc-466a-aee7-e213799f0241)

Reply:
Looks really good! Nice work!